packages/org-chart/src/orgChart.ts

```diff
@@ -1029,1227 +1040,653 @@ function createSpecialColumnLines(
-            } else {
-              className = ` ${lineUp}-${indexInRow === 2 ? 'right-half' : indexInRow === 3 ? 'left-half' : 'left'} ${lineUp}`;
-            }
-          } else if (specialColumnsList.includes(2)) {
-            if (isLaptop) {
-              className = ` ${lineUp}-${indexInRow === 1 || indexInRow === 4 ? 'right-half' : indexInRow === 3 ? 'right' : 'left-half'} ${lineUp}`;
-            } else {
-              className = ` ${lineUp}-${indexInRow === 1 ? 'right' : indexInRow === 3 ? 'left-half' : 'left'} ${lineUp}`;
-            }
-          } else if (specialColumnsList.includes(3)) {
-            if (isLaptop) {
-              className = ` ${lineUp}-${indexInRow === 1 || indexInRow === 4 ? 'right' : 'right-half'} ${lineUp}`;
-            } else {
-              className = ` ${lineUp}-${indexInRow === 1 ? 'right' : indexInRow === 2 ? 'right-half' : 'left'} ${lineUp}`;
-            }
-          } else if (specialColumnsList.includes(4)) {
-            if (isLaptop) {
-              className = ` ${lineUp}-${indexInRow === 1 ? 'right' : indexInRow === 2 ? 'right-half' : 'left-half'} ${lineUp}`;
-            } else {
-              className = ` ${lineUp}-${indexInRow === 3 ? 'right-half' : indexInRow === 5 ? 'left' : 'right'} ${lineUp}`;
-            }
-          } else if (specialColumnsList.includes(5)) {
-            className = ` ${lineUp}-${indexInRow === 3 ? 'right-half' : indexInRow === 4 ? 'left-half' : 'right'} ${lineUp}`;
-          }
-        }
+        let upperHalfHighest = findHighestChildrenAmountInRow(row, 0, 1);
+        let lowerHalfHighest = findHighestChildrenAmountInRow(row, 2, 2);
+        diff = upperHalfHighest - lowerHalfHighest;
       }
     }
-    return className;
   }
+  return diff.toString();
 }
 
-// create a that takes a number, and return true if the number is even or 1, false if the number is odd
-function isEvenOrOne(num: number) {
-  return num % 2 === 0 || num === 1;
+function getHighestNumber(num1: number, num2: number) {
+  return num1 >= num2 ? num1 : num2;
 }
 
-function isOdd(number: number) {
-  return number % 2 !== 0;
+function findHighestChildrenAmountInRow(
+  row: Row,
+  indexStart: number,
+  indexEnd: number,
+) {
+  let highest = 0;
+  for (let i = indexStart; i <= indexEnd; i++) {
+    // Use optional chaining to safely access nested properties
+    const childrenLength = row.row[i]?.children?.length ?? 0;
+    if (childrenLength > highest) {
+      highest = childrenLength;
+    }
+  }
+
+  return highest;
 }
 
-function getBreakpointName(width: number) {
-  let breakpointName =
-    width > 1500
-      ? 'main'
-      : width > 992
-        ? 'laptop'
-        : width > 768
-          ? 'tablet'
-          : 'mobile';
-  return breakpointName;
+function createTOC(toc: TableOfContentsItem[], isMobile?: boolean) {
+  const tocBox = createElement('ul');
+  tocBox.classList.add(`${prefix}-toc-box`);
+  isMobile && tocBox.classList.add(`${prefix}-toc-box-mobile`);
```

Comment:
```suggestion
  tocBox.classList.add(`${prefix}-toc-box`, isMobile && `${prefix}-toc-box-mobile`);
```

`classList.add` takes a comma seperated list of strings, so we can add the `isMobile` check on the same line.

Reply:
Found out this does not work, because it will "throwa a "SyntaxError" DOMException if one of the arguments is the empty string."